credentials/credentials.go

```diff
@@ -92,7 +92,7 @@ type CommonAuthInfo struct {
 }
 
 // GetCommonAuthInfo returns the pointer to CommonAuthInfo struct.
-func (c *CommonAuthInfo) GetCommonAuthInfo() *CommonAuthInfo {
+func (c CommonAuthInfo) GetCommonAuthInfo() CommonAuthInfo {
```

Comment:
This seems a little weird to have a method on a type which returns itself. Was this added initially (with a pointer receiver) because we could not extend the `AuthInfo` interface due to backward compatibility reasons?

Reply:
Yes, I believe that's the reason.